x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceService.java

```diff
@@ -170,135 +156,9 @@ private static Map<String, DefaultModelConfig> initDefaultEndpoints(
         );
     }
 
-    private record DefaultModelConfig(Model model, MinimalServiceSettings settings) {}
```

Comment:
Basically moved all this logic to the new `ElasticInferenceServiceAuthorizationHandler` class.